src/logic/nontrivial.lean

```diff
@@ -209,8 +220,19 @@ begin
   dec_trivial
 end
 ```
+
+```
+def myeq {α : Type} (a b : α) : Prop := a = b
+
+example {α : Type} (a b : α) (h : a = b) : myeq a b :=
+begin
+  success_if_fail { nontriviality α }, -- Fails
+  nontriviality α [myeq], -- There is no `nontrivial α` hypothesis available
+  assumption
+end
+```
 -/
-meta def nontriviality (t : parse parser.pexpr?) : tactic unit :=
+meta def nontriviality (t : parse parser.pexpr?) (lems : parse simp_arg_list) : tactic unit :=
```

Comment:
There's a decision to be made about precedences here, since a list of simp args following a `pexpr` is ambiguous with just a `pexpr`. What you have now gives low precedence to the `pexpr`, so you need parentheses in:

```lean
example (α : ℕ → Type) (a b : α 0) : a = b :=
begin 
  nontriviality (α 0) [nat.zero_lt_one],
end
```
Alternatively you can change things to `nontriviality α 0 using [nat.zero_lt_one]`.

I can't think of an existing tactic that does exactly the latter (there probably is one), but if you prefer that I can do it/explain how.

Reply:
I found an example in the `mono` tactic. Done.